aepsych/models/inducing_point_allocators.py

```diff
@@ -0,0 +1,303 @@
+            name (str, optional): Name of the allocator, defaults to None.
+            options (Dict[str, Any], optional): Additional options, defaults to None.
+
+        Returns:
+            SobolAllocator: A SobolAllocator instance.
+        """
+        return cls(**cls.get_config_options(config, name, options))
+
+    @classmethod
+    def get_config_options(
+        cls,
+        config: Config,
+        name: Optional[str] = None,
+        options: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """Get configuration options for the SobolAllocator.
+
+        Args:
+            config (Config): Configuration object.
+            name (str, optional): Name of the allocator, defaults to None.
+            options (Dict[str, Any], optional): Additional options, defaults to None.
+
+        Returns:
+            Dict[str, Any]: Configuration options for the SobolAllocator.
+        """
+        if name is None:
+            name = cls.__name__
+        bounds = config.gettensor(name, "bounds")
+        return {"bounds": bounds}
+
+
+class KMeansAllocator(InducingPointAllocator, ConfigurableMixin):
+    """An inducing point allocator that uses k-means++ to allocate inducing points."""
+
+    def __init__(self) -> None:
+        """Initialize the KMeansAllocator."""
+        super().__init__()
+
+    def _get_quality_function(self) -> None:
+        """K-means++ does not require a quality function, so this returns None."""
+        return None
+
+    def allocate_inducing_points(
+        self,
+        inputs: torch.Tensor,
+        covar_module: Optional[torch.nn.Module] = None,
+        num_inducing: int = 10,
+        input_batch_shape: torch.Size = torch.Size([]),
+    ) -> torch.Tensor:
+        """
+        Generates `num_inducing` inducing points using k-means++ initialization on the input data.
+
+        Args:
+            inputs (torch.Tensor): A tensor of shape (n, d) containing the input data.
+            covar_module (torch.nn.Module, optional): Kernel covariance module; included for API compatibility, but not used here.
+            num_inducing (int, optional): The number of inducing points to generate. Defaults to 10.
+            input_batch_shape (torch.Size, optional): Batch shape, defaults to an empty size; included for API compatibility, but not used here.
+
+        Returns:
+            torch.Tensor: A (num_inducing, d)-dimensional tensor of inducing points selected via k-means++.
+        """
+        # Ensure inputs are unique to avoid duplication issues with k-means++
+        unique_inputs = torch.unique(inputs, dim=0)
+
+        # If unique inputs are less than or equal to the required inducing points, return them directly
+        if unique_inputs.shape[0] <= num_inducing:
+            return unique_inputs
+
+        # Run k-means++ on the unique inputs to select inducing points
+        inducing_points = torch.tensor(
+            kmeans2(unique_inputs.numpy(), num_inducing, minit="++")[0],
```

Comment:
`unique_inputs.numpy()` fails on GPU. Need to force it to cpu first.
```suggestion
            kmeans2(unique_inputs.cpu().numpy(), num_inducing, minit="++")[0],
```